pkg/chain/chain.go

```diff
@@ -30,16 +30,18 @@ type Chain interface {
 }
 
 type chain struct {
-	handlers []ChainHandler
-	final    http.Handler
-	status   *http.Handler
+	handlers       []ChainHandler
+	final          http.Handler
+	status         *http.Handler
+	sourceIPHeader string
 }
 
 // New creates a new chain.
-func New(final http.Handler, handlers ...ChainHandler) Chain {
+func New(final http.Handler, sourceIPHeader string, handlers ...ChainHandler) Chain {
```

Comment:
For this, I think it would be better to have a ip extractor interface where:
 - if there is a header configured, the ip extractor will extract the IP from the headers
 - otherwise, the request's ip will be used.

WDYT ?

Reply:
Where would you see this interface going? By interface there, you mean an actual code interface that would be implemented by only one class and used once or an "interface" as a decoupled way of getting the IP? For the latter, would you put this outside the chain handler?

I'm open to streamline this more but I guess I need a bit more context on what you think for the IP extractor.